### ISSUE TITLE
Training loss spikes and validation metrics collapse after learning rate warmup

### ISSUE THREAD

**Issue by shaokai1209** [SofieneKammoun/SE-NAC-CODE#3]
I noticed that training is stable for the first few epochs, but then **collapses suddenly right after the learning rate warmup phase ends** (around epoch 10). The training loss spikes and validation metrics (SI-SDR, ESTOI) drop to very poor values.

I suspect it might be related to the `lr_scheduler` setup:
```python
lr_scheduler = torch.optim.lr_scheduler.LambdaLR(
    optimizer,
    lr_lambda=lambda epoch: cosine_decay(epoch, 10, NUM_EPOCHS)
)

Have you observed anything similar in your own runs, or do you have suggestions for making the training more robust across different datasets?
